code/modules/projectiles/guns/energy.dm

```diff
@@ -407,9 +334,58 @@
 
 	in_chamber = get_ammo_object(chamber_items[current_chamber_position])
 
-/obj/item/weapon/gun/energy/lasgun/lasrifle/update_item_state(mob/user) //Without this override icon states for wielded guns won't show. because lasgun overrides and this has no charge icons
-	item_state = "[initial(icon_state)][flags_item & WIELDED ? "_w" : ""]"
+//Tesla gun
+/obj/item/weapon/gun/energy/lasgun/lasrifle/tesla
+	name = "\improper Terra Experimental tesla shock rifle"
+	desc = "A Terra Experimental energy rifle that fires balls of elecricity that shock all those near them, it is meant to drain the plasma of unidentified creatures from within, limiting their abilities. Unlike the other TE Laser weapons, lasers don't come out of this weird weapon. As with all TE Laser weapons, they use a lightweight alloy combined without the need for bullets any longer decreases their weight and aiming speed quite some vs their ballistic counterparts. Uses standard Terra Experimental  (abbreviated as TE) power cells."
```

Comment:
```suggestion
	desc = "A Terra Experimental energy rifle that fires balls of elecricity that shock all those near them, it is meant to drain the plasma of unidentified creatures from within, limiting their abilities. As with all TE Laser weapons, they use a lightweight alloy combined without the need for bullets any longer decreases their weight and aiming speed quite some vs their ballistic counterparts. Uses standard Terra Experimental (TE) power cells."
```
it's as if `A Terra Experimental energy rifle that fires balls of elecricity that shock all those near them` isn't enough